xc/common/utils/prjxray_create_synth_tiles.py

```diff
@@ -70,33 +179,50 @@ def main():
 
     with DatabaseCache(args.connection_database, read_only=True) as conn:
         synth_tiles['info'] = j['info']
-        vbrk_in_use = set()
+        tile_in_use = set()
         for port in j['ports']:
-            if port['name'].startswith('dout['):
-                port_type = 'input'
-                is_clock = False
-            elif port['name'].startswith('din['):
-                is_clock = False
-                port_type = 'output'
-            elif port['name'].startswith('clk'):
-                port_type = 'output'
-                is_clock = True
+            if args.roi:
```

Comment:
I'm wondering if the logic here can be unified to use your method, rather than having a giant `if` here?

For example, have the wire specified, and have the code go from wire -> node, and pick a suitable location.

I like having an explicitly type, why not propigate that to the ROI definition files?

Reply:
We can do that too.  Just was trying to limit how much I touched regular roi stuff, but if you're ok with it I can propagate to normal ROI definitions.

Reply:
> For example, have the wire specified, and have the code go from wire -> node, and pick a suitable location.

I just looked, and the current ROI format specifies both the wire and the node.  Maybe simply have the logic go "if wire is not specified, pick a wire, otherwise check if the specified wire is legal based on the selection logic".  That enables both automatic and fine grained control, if needed.

Reply:
> Just was trying to limit how much I touched regular roi stuff

I'd rather update the ROI stuff than maintain two extremely similar code paths.

Reply:
Ok that sounds good.